client/blocks/image-selector/index.jsx

```diff
@@ -0,0 +1,173 @@
+export class ImageSelector extends Component {
+	static propTypes = {
+		compact: PropTypes.bool,
+		hasDropZone: PropTypes.bool,
+		imageIds: PropTypes.array,
+		isDropZoneVisible: PropTypes.bool,
+		maxWidth: PropTypes.number,
+		onAddImage: PropTypes.func,
+		onImageChange: PropTypes.func,
+		onImageSelected: PropTypes.func,
+		onRemoveImage: PropTypes.func,
+		selecting: PropTypes.bool,
+		showEditIcon: PropTypes.bool,
+		site: PropTypes.object,
+		siteId: PropTypes.number,
+		translate: PropTypes.func,
+	};
+
+	static defaultProps = {
+		compact: false,
+		hasDropZone: false,
+		imageIds: [],
+		isDropZoneVisible: false,
+		maxWidth: 450,
+		onAddImage: noop,
+		onImageSelected: noop,
+		onImageChange: noop,
+		onRemoveImage: noop,
+	};
+
+	state = {
+		isSelecting: false,
+	};
+
+	showMediaModal = () => {
+		const { siteId, imageIds } = this.props;
+
+		if ( imageIds ) {
+			const images = imageIds.map( imageId => MediaStore.get( siteId, imageId ) );
+			MediaActions.setLibrarySelectedItems( siteId, images );
+		}
+
+		this.setState( {
+			isSelecting: true,
+		} );
+	};
+
+	hideMediaModal = () => {
+		this.setState( {
+			isSelecting: false,
+		} );
+	};
+
+	setImage = value => {
+		if ( value ) {
+			this.props.onImageSelected( value );
+		}
+		this.hideMediaModal();
+	};
+
+	removeImage = image => {
+		this.props.onRemoveImage( image );
+	};
+
+	addImage = image => {
+		this.props.onAddImage( image );
+	};
+
+	// called when media library item transitions from temporary ID to a permanent ID, e.g.,
+	// after creating an item by uploading or selecting from Google library.
+	onImageChange = images => {
+		this.props.onImageChange( images );
+	};
+
+	renderMediaModal() {
+		if ( ! this.props.siteId ) {
+			return;
+		}
+
+		const { multiple, selecting, siteId, translate } = this.props;
+		const { isSelecting } = this.state;
+
+		return (
+			<MediaLibrarySelectedData siteId={ siteId }>
+				<MediaModal
+					visible={ selecting || isSelecting }
+					onClose={ this.setImage }
+					siteId={ siteId }
+					labels={ { confirm: multiple ? translate( 'Set images' ) : translate( 'Set image' ) } }
+					enabledFilters={ [ 'images' ] }
+					galleryViewEnabled={ false }
+					{ ...! multiple && { single: true } }
+				/>
+			</MediaLibrarySelectedData>
+		);
+	}
+
+	renderSelectedImages() {
+		const { compact, siteId, imageIds, maxWidth, multiple, showEditIcon } = this.props;
+
+		return (
+			<ImageSelectorPreview
+				siteId={ siteId }
+				itemIds={ imageIds }
+				maxWidth={ maxWidth }
+				onImageChange={ this.onImageChange }
+				onImageClick={ this.showMediaModal }
+				onRemoveImage={ this.removeImage }
+				multiple={ multiple }
+				compact={ compact }
+				showEditIcon={ showEditIcon }
+			/>
+		);
+	}
+
+	render() {
+		const { imageIds } = this.props;
+		const classes = classnames( 'image-selector', {
+			'is-assigned': !! imageIds && imageIds.length,
+			'has-active-drop-zone': this.props.hasDropZone && this.props.isDropZoneVisible,
```

Comment:
Minor but you could add `hasDropZone` and `isDropZoneVisible` to the deconstructions above.